internal/testplanet/uplink.go

```diff
@@ -198,128 +192,124 @@ func (uplink *Uplink) UploadWithExpirationAndConfig(ctx context.Context, satelli
 		}
 	}
 
-	metainfo, streams, cleanup, err := DialMetainfo(ctx, uplink.Log.Named("metainfo"), config, uplink.Identity)
+	project, bucket, err := planetUplink.GetProjectAndBucket(ctx, satellite, bucketName, config)
 	if err != nil {
 		return err
 	}
-	defer func() {
-		err = errs.Combine(err, cleanup())
-	}()
+	defer func() { err = errs.Combine(err, project.Close(), bucket.Close()) }()
```

Comment:
nit: not sure if that will make a difference but currently project will be closed before bucket and we should close things in reverse open order.

Reply:
good idea, i will fix